neonlabsorg/neon-evm
# #313 Migrate accounts v1 -> v2

Created instruction `MigrateAccount` and corresponding cli command `migrate-account`.

**How to test it manually in a local environment**

— Build the neon-cli locally (branch `313-migrate-accounts-v2`)
— Start our set of containers from `proxy/docker-compose-test.yml` (`develop` branch) as usual
— Copy the new version of `neon-cli` and the supplementary script `neon-accounts.py`:
``` 
docker cp neon-evm/evm_loader/target/release/neon-cli proxy:/spl/bin/neon-cli-v2
docker cp neon-evm/evm_loader/utils/neon-accounts.py proxy:/spl/bin/
```
— Enter the proxy container: `docker exec -it proxy bash`
— The following should be made inside the proxy container:

```
export SOLANA_URL=http://solana:8899
export FAUCET_URL=http://faucet:3333
export EVM_LOADER=53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io
export EVM_REVISION=<id-of-new-docker-evm-loader-image> (CI logs should report the id)

# Create one or more V1 accounts
curl -i -X POST $FAUCET_URL/request_neon -d '{ "wallet": "0x1000000000000000000000000000000000000000", "amount": 10 }'

# Make sure it exists
neon-accounts.py

# Make sure the associated token accounts have been created as well
spl-token account-info 4pkUvFFstx6VdqjTsjwsx4ko7MXN82eEwr7ZUCQuygkL

# Remove the old container evm_loader
docker container rm $(docker container ps -aq --filter name=evm_loader)

# Deploy the new evm_loader
docker run --network=container:proxy -e SOLANA_URL=http://solana:8899 --entrypoint=deploy-evm.sh --name=evm_loader neonlabsorg/evm_loader:$EVM_REVISION

# Create throwaway ether account to init the V2 pool account properly
neon-cli-v2 create-ether-account 0x9999999999999999999999999999999999999999 --url $SOLANA_URL --evm_loader $EVM_LOADER

# Migration should work now
neon-cli-v2 migrate-account 0x1000000000000000000000000000000000000000 --url $SOLANA_URL --evm_loader $EVM_LOADER

# Make sure the accounts are now V2
neon-accounts.py

# Make sure the associated token accounts are deleted
spl-token account-info 4pkUvFFstx6VdqjTsjwsx4ko7MXN82eEwr7ZUCQuygkL
```

**How to perform mass migration of all available V1 ether accounts**
```
neon-accounts.py migrate
```
